src/Utilities/TypeTraits.hpp

```diff
@@ -373,6 +373,29 @@ using remove_cvref_t = typename remove_cvref<T>::type;
 /// \ingroup TypeTraitsGroup
 /// A collection of useful type traits
 namespace tt {
+// @{
+/*!
+ * \ingroup TypeTraitsGroup
+ * \brief Check if `T` is copy constructible
+ *
+ * The STL `std::is_copy_constructible` does not work correctly with some types,
+ * such as `std::unordered_map` (likely a problem with `std::unordered_map`).
```

Comment:
`std::is_copy_constructible` doesn't do what you want (which would generally be the more useful behavior), but it does what the standard says it should do (as do the containers).  Please name this something else to avoid confusion and document the difference from `std::is_copy_constructible`.

Reply:
Do you have a suggestion for an alternative name? I've be trying to think of something that doesn't suck and it's been a struggle. Maybe `can_be_copy_constructed`?